consensus/hotstuff/verification/staking_verifier.go

```diff
@@ -92,7 +92,54 @@ func (v *StakingVerifier) VerifyQC(signers flow.IdentityList, sigData []byte, bl
 	}
 
 	if !stakingValid {
-		return fmt.Errorf("invalid aggregated staking sig for block %v: %w", block.BlockID, model.ErrInvalidSignature)
+		return fmt.Errorf("invalid aggregated staking sig for block %v: %w", blockID, model.ErrInvalidSignature)
+	}
+	return nil
+}
+
+// VerifyTC checks cryptographic validity of the TC's `sigData` for the given view.
+// It is the responsibility of the calling code to ensure
+// that all `voters` are authorized, without duplicates. Return values:
+//  - nil if `sigData` is cryptographically valid
+//  - model.ErrInvalidFormat if `sigData` has an incompatible format
+//  - model.ErrInvalidSignature if a signature is invalid
+//  - unexpected errors should be treated as symptoms of bugs or uncovered
+//	  edge cases in the logic (i.e. as fatal)
+func (v *StakingVerifier) VerifyTC(voters flow.IdentityList, sigData []byte, view uint64, highQCViews []uint64) error {
+	return verifyTC(voters, sigData, view, highQCViews, v.stakingHasher)
+}
+
+// verifyTC checks cryptographic validity of the TC's `sigData` for the given view.
+// It is the responsibility of the calling code to ensure
+// that all `voters` are authorized, without duplicates. Return values:
+//  - nil if `sigData` is cryptographically valid
+//  - model.ErrInvalidFormat if `sigData` has an incompatible format
```

Comment:
Agree about `IsInvalidInputsError`, was thinking the crypto method would return that for invalid `sigData` but seems not to be the case.

I think the specific code change suggestion still applies though. In particular, we do not return `ErrInvalidFormat` if `sigData` has an incompatible format

Reply:
what do you mean by incompatible format? Incompatible format can happen only in case of `CombinedVerifier` where we have two sigs instead of one.

Reply:
What I mean is the documentation and implementation don't match. 

The function documentation says:
> Return values:
> - model.ErrInvalidFormat if `sigData` has an incompatible format

But in the implementation, we do:
```go
if len(voters) == 0 {
	return model.NewInvalidFormatErrorf("empty list of signers")
}
if len(voters) != len(highQCViews) {
	return model.NewInvalidFormatErrorf("signers and highQCViews mismatch")
}
```

My suggestion was to change the documentation to:
> - model.ErrInvalidFormat if `voters`/`highQCViews` have differing lengths, or are empty

because that is what we do in the implementation.